docs/src/contributing.md

```diff
@@ -0,0 +1,159 @@
+
+```
+   pick 01d1124 <commit message 1>
+   pick 6340aaa <commit message 2>
+   pick ebfd367 <commit message 3>
+   pick 30e0ccb <commit message 4>
+
+   # Rebase 60709da..30e0ccb onto 60709da
+   #
+   # Commands:
+   #  p, pick = use commit
+   #  e, edit = use commit, but stop for amending
+   #  s, squash = use commit, but meld into previous commit
+   #
+   # If you remove a line here THAT COMMIT WILL BE LOST.
+   # However, if you remove everything, the rebase will be aborted.
+##
+```
+
+We want to keep the first commit and squash the last 3. We do so by changing the last three commits to `squash` and then do `:wq` on vim.
+
+```
+   pick 01d1124 <commit message 1>
+   squash 6340aaa <commit message 2>
+   squash ebfd367 <commit message 3>
+   squash 30e0ccb <commit message 4>
+
+   # Rebase 60709da..30e0ccb onto 60709da
+   #
+   # Commands:
+   #  p, pick = use commit
+   #  e, edit = use commit, but stop for amending
+   #  s, squash = use commit, but meld into previous commit
+   #
+   # If you remove a line here THAT COMMIT WILL BE LOST.
+   # However, if you remove everything, the rebase will be aborted.
+```
+
+Then in the next screen that appears, we can just delete all messages that
+we do not want to show in the commit. After this is done and we are back to 
+the console, we have to force push. We need to force push because we rewrote
+the local commit history.
+
+```
+$ git push -u origin <name_of_local_branch> --force
+```
+
+You can find more information about squashing [here](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#squash-your-changes).
+
+### Unit testing
+
+Currently a number of checks are run per commit for a given PR.
+
+- `JuliaFormatter` checks if the PR is formatted with `.dev/climaformat.jl`.
+- `Documentation` rebuilds the documentation for the PR and checks if the docs
+  are consistent and generate valid output.
+- `Unit Tests` run subsets of the unit tests defined in `tests/`, using `Pkg.test()`.
+  The tests are run in parallel to ensure that they finish in a reasonable time.
+  The tests only run the latest commit for a PR, branch and will kill any stale jobs on push.
+  These tests are only run on linux (Ubuntu LTS).
+
+Unit tests are run against every new commit for a given PR,
+the status of the unit-tests are not checked during the merge
+process but act as a sanity check for developers and reviewers.
+Depending on the content changed in the PR, some CI checks that
+are not necessary will be skipped.  For example doc only changes
+do not require the unit tests to be run.
+
+### The merge process
+
+We use [`bors`](https://bors.tech/) to manage merging PR's in the the `EnsembleKalmanProcesses` repo.
+If you're a collaborator and have the necessary permissions, you can type
+`bors try` in a comment on a PR to have integration test suite run on that
+PR, or `bors r+` to try and merge the code.  Bors ensures that all integration tests
```

Comment:
"The integration tests currently runs example cases in `examples/`. Any breaking changes will need to also update the `examples/`, else `bors` will fail "